js2-imenu-extras.el

```diff
@@ -221,6 +229,161 @@ Currently used for jQuery widgets, Dojo and Enyo declarations."
                                      (list name-value))
                                    (js2-node-abs-pos methods))))))
 
+(defun js2-imenu-record-mocha-describe ()
+  "Populate `js2-imenu-recorder' with mocha-like describe/it/beforeEach/… nodes."
+  (let ((node (js2-node-at-point (1- (point)))))
+    (when (js2-imenu-extras--mocha-top-level-describe-p node)
+      (js2-imenu-extras--mocha-visit-node node (list)))))
+
+(defun js2-imenu-extras--mocha-visit-node (node qname)
+  "Search NODE and its children for mocha test blocks.
+
+If mocha test blocks are found (e.g., a describe() or it() block)
+they are added to `js2-imenu-recorder' with QNAME as prefix.
+
+QNAME is a list of nodes representing the qualified name of
+NODE's parent.  If NODE has no parent, QNAME is the empty list.
+The last item of QNAME is NODE's parent name while the item
+before that is NODE's grandparent name etc."
+  (js2-visit-ast
+   node
+   (lambda (child end-p)
+     (when (not end-p)
+       (js2-imenu-extras--mocha-check-unknown-node child qname)))))
+
+(defun js2-imenu-extras--mocha-check-unknown-node (node qname)
+  "If NODE is a mocha test block, populate `js2-imenu-recorder'.
+
+QNAME is a list of nodes representing the qualified name of
```

Comment:
The description of QNAME is repeated multiple times in other functions, the whole paragraph. Appreciate the good documentation, but it looks a little repetitive.

How about we keep it in `js2-imenu-extras--mocha-visit-node`, but in other functions say

    QNAME is the same as described in `js2-imenu-extras--mocha-visit-node'.

?